src/client/src/consumer.rs

```diff
@@ -48,15 +48,15 @@ use crate::spu::SpuPool;
 pub struct PartitionConsumer {
     topic: String,
     partition: i32,
-    pool: Arc<SpuPool>,
+    inner: Arc<FluvioInner>,
```

Comment:
Consumer should not directly reference Fluvio which is connection to SC after looking up.  Instead we should create wrapper for SpuPool which maintain ref to Fluvio

Reply:
Well, part of the reason that this is needed is because the `SharedAllMultiplexerSocket` passed into `SpuPool` isn't kept. So, when the functions return, the `socket : Arc<AllMultiplexerSocket>`s go [out of scope in something like `SpuPool::start`](https://github.com/infinyon/fluvio/blob/51360927a2efc067d7015c3a186028b8300cc3d3/src/client/src/spu/spu_pool.rs#L63-L74) they drop which causes the [`self.terminate.notify` call in `MultiplexerSocket::drop()`](https://github.com/infinyon/fluvio-socket/blob/f537ef106530f74a2d0f9a772e9594dffc7f731f/crates/socket/src/multiplexing.rs#L81) which is an event to [the `MultiPlexingResponseDispatcher` to close the socket.](https://github.com/infinyon/fluvio-socket/blob/f537ef106530f74a2d0f9a772e9594dffc7f731f/crates/socket/src/multiplexing.rs#L357)

Another way to do this is to pass in owned instances of `SharedAllMultiplexerSocket`s rather than [`&AllMultiplexerSockets`](https://github.com/infinyon/fluvio/blob/51360927a2efc067d7015c3a186028b8300cc3d3/src/client/src/spu/spu_pool.rs#L65).

@nicholastmosher Thoughts?